R/Figures.R

```diff
@@ -82,13 +82,104 @@ TADA_boxplot <- function(filtered.data) {
       yaxis = list(title = y_label, titlefont = list(size = 16, family = "Arial"), tickfont = list(size = 16, family = "Arial"),
                    hoverformat = ',.4r', linecolor = "black", rangemode = 'tozero', 
                    showgrid = FALSE, tickcolor= "black"),
-      hoverlabel=list(bgcolor="white")
+      hoverlabel=list(bgcolor="white"),
+      title = paste0("Boxplot of ", char, " (", unit, ")"), 
+      plot_bgcolor = "#e5ecf6"
     ) %>% 
     plotly::config(displayModeBar = FALSE)
   
   return(base_boxplot)
 }
 
+#' Create Histogram
```

Comment:
These plots look really great!! My only suggestion would be to remove the dependency of the histogram and boxplot on the TADA.ComparableDataIdentifier, and instead create another input for this plot that is something along the lines of id_col = comparables column OR characteristic column. I could see a user wanting to visualize data prior to running the harmonization step, so it would be nice to give them the option to just filter to a specific characteristic name. You could make the documentation say that id_col can equal either TADA.DataComparableIdentifier OR TADA.CharacteristicName, and then it can make sure that the one the user picked is in the dataset. Currently, it is also problematic that it requires one TADA.ComparableIdentifier but then uses the TADA.CharcteristicName, which may not be unique to the dataset (since the comparable identifier allows users to group multiple characteristic names into one group).

Reply:
ooh good point! hmm, I had originally created the boxplot and histogram functions under the assumption that it would be required that the datasets be harmonized in case units of measure were not consistent across one characteristic. Do you think that is not usually an issue across one characteristic? Or should I just create an error/warning if there is more than one unit of measure?

Reply:
Oooh also a good point. Shoot. Yeah, I would probably just create a warning if there are multiple units (in any situation, but I guess would only happen if id_col = TADA.CharName), multiple characteristic names (if id_col = TADA.CharacteristicName), or multiple comparable data identifiers (if id_col = "TADA.ComparableDataIdentifier"), and then make sure the plot labels reflect the id_col input. I think the vision is to leverage these in TADAShiny, too, and in that interface, we can make users pick a unique characteristic + unit/comparable combination to be fed into the plotting functions. I'm not exactly sure the full implementation of the harmonization template in the QC module, and how much of it will be user-populated versus TADA-suggested (currently, we have some nutrient-related suggestions with everything else having a TADA.ComparableDataIdentifier of NA and I recall Cristina wanted to do some more work with it), but I think making the plots more flexible to be used throughout the process would be a nice addition. What do you think?

Reply:
that sounds like a good idea! I'll add the input column options and unit error message :)

Reply:
Sounds good, thanks!